### PR TITLE
Implement extraction of inline examples and playground link

### PR DESCRIPTION
**Pull request by tharders**

### CODE REVIEW

File: pages/extensions/amp_example_preview/util/example_document.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    self.preview = export_attributes.get('preview', PREVIEW_NONE)
+    if self.preview == PREVIEW_INLINE and (self.head or self.custom_style)\
+      or self.preview not in PREVIEW_MODES:
+      # no inline preview supported if head elements are present
+      self.preview = PREVIEW_NONE
+
+    self.playground = 'playground' not in export_attributes or export_attributes['playground'].lower() != 'false'
+
+  def has_amp_script(self):
+    return self.has_head_tag('style', 'amp[\da-z]*-boilerplate')
```
Comment (matthiasrohmer) [inline pages/extensions/amp_example_preview/util/example_document.py]: This was probably intended to check for `v0.js`?

File: pages/extensions/amp_example_preview/util/example_document.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  def has_boilerplate(self):
+    return self.has_head_tag('style', 'amp[\da-z]*-boilerplate')
+
+  def has_import_in_head(self, dependency):
+    if not self.head:
+      return False
+    regex = re.compile(r'<script(?:\s[^>]*)?\scustom-[a-z]+\s*=\s*"?{dependency}[\s">]'.format(dependency=dependency))
+    return regex.search(self.head) is not None
+
+  def has_head_tag(self, tag_name, attribute_name_re='', attribute_value_re=''):
```
Comment (matthiasrohmer) [inline pages/extensions/amp_example_preview/util/example_document.py]: Only my opinion: `has_tag_in_head` might be more fitting - my first assumption was this method checks for existance of `<head>` 🙂 - would also fit the above method `has_import_in_head`.

File: pages/extensions/amp_example_preview/util/markdown_processor.py [hunk 3]
```diff
@@ -1,73 +1,67 @@
-import os
-from grow.documents import document
-from .example_extractor import *
-from .preview import ExamplePreview
-from .constants import *
-
-import os
+from example_extractor import *
```
Comment (matthiasrohmer) [inline pages/extensions/amp_example_preview/util/markdown_processor.py]: I see it's easier to write, but can we instead use named imports here? 😉 That would make it more obvious where things are coming from.

File: pages/extensions/amp_example_preview/util/preview.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import re
 import json
-from xml.sax.saxutils import escape, unescape
+from cgi import escape  # we use cgi.escape since we do not want to escape any quotes
```
Comment (matthiasrohmer) [inline pages/extensions/amp_example_preview/util/preview.py]: The more you know 👍

File: pages/extensions/amp_example_preview/util/example_document.py (hunk 1 above)
Comment (tharders) [reply pages/extensions/amp_example_preview/util/example_document.py]: Ooops, good that you spotted this one! Of course you are right.

File: pages/extensions/amp_example_preview/util/example_document.py (hunk 2 above)
Comment (tharders) [reply pages/extensions/amp_example_preview/util/example_document.py]: Yep, much better!

File: pages/extensions/amp_example_preview/util/markdown_processor.py (hunk 3 above)
Comment (tharders) [reply pages/extensions/amp_example_preview/util/markdown_processor.py]: True :-)

File: pages/extensions/amp_example_preview/standalone_example_validator.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+  def get_formats_from_default_language(self, path):
+    parent_path = path[:path.rindex('@')] + '.md'
+    try:
+      content = self.load_file_content(parent_path)
+      return self.get_formats_from_frontmatter(content, parent_path)
+    except:
+      sys.stdout.write('Unable to load default language content file at {}\n'.format(parent_path))
+      return [DEFAULT_FORMAT]
+
+  def createTempDoc(self, formats):
```
Comment (matthiasrohmer) [inline pages/extensions/amp_example_preview/standalone_example_validator.py]: Really just a nit: `create_temp_doc`

**Pull request merged**